Sources/SwiftSyntax/generated/SyntaxCollections.swift

```diff
@@ -1675,113 +1652,13 @@ public struct TupleTypeElementListSyntax: SyntaxCollection, SyntaxHashable {
-      } else if let node = node.as(LifetimeTypeSpecifierSyntax.self) {
-        self = .lifetimeTypeSpecifier(node)
-      } else {
-        return nil
-      }
-    }
-
-    public static var structure: SyntaxNodeStructure {
-      return .choices([.node(SimpleTypeSpecifierSyntax.self), .node(LifetimeTypeSpecifierSyntax.self)])
-    }
-
-    /// Checks if the current syntax node can be cast to ``SimpleTypeSpecifierSyntax``.
-    ///
-    /// - Returns: `true` if the node can be cast, `false` otherwise.
-    public func `is`(_ syntaxType: SimpleTypeSpecifierSyntax.Type) -> Bool {
-      return self.as(syntaxType) != nil
-    }
-
-    /// Attempts to cast the current syntax node to ``SimpleTypeSpecifierSyntax``.
-    ///
-    /// - Returns: An instance of ``SimpleTypeSpecifierSyntax``, or `nil` if the cast fails.
-    public func `as`(_ syntaxType: SimpleTypeSpecifierSyntax.Type) -> SimpleTypeSpecifierSyntax? {
-      return SimpleTypeSpecifierSyntax.init(self)
-    }
-
-    /// Force-casts the current syntax node to ``SimpleTypeSpecifierSyntax``.
-    ///
-    /// - Returns: An instance of ``SimpleTypeSpecifierSyntax``.
-    /// - Warning: This function will crash if the cast is not possible. Use `as` to safely attempt a cast.
-    public func cast(_ syntaxType: SimpleTypeSpecifierSyntax.Type) -> SimpleTypeSpecifierSyntax {
-      return self.as(SimpleTypeSpecifierSyntax.self)!
-    }
-
-    /// Checks if the current syntax node can be cast to `LifetimeTypeSpecifierSyntax`.
-    ///
-    /// - Returns: `true` if the node can be cast, `false` otherwise.
-    /// - Note: Requires experimental feature `nonescapableTypes`.
-    #if compiler(>=5.8)
-    @_spi(ExperimentalLanguageFeatures)
-    #endif
-    public func `is`(_ syntaxType: LifetimeTypeSpecifierSyntax.Type) -> Bool {
-      return self.as(syntaxType) != nil
-    }
-
-    /// Attempts to cast the current syntax node to `LifetimeTypeSpecifierSyntax`.
-    ///
-    /// - Returns: An instance of `LifetimeTypeSpecifierSyntax`, or `nil` if the cast fails.
-    /// - Note: Requires experimental feature `nonescapableTypes`.
-    #if compiler(>=5.8)
-    @_spi(ExperimentalLanguageFeatures)
-    #endif
-    public func `as`(_ syntaxType: LifetimeTypeSpecifierSyntax.Type) -> LifetimeTypeSpecifierSyntax? {
-      return LifetimeTypeSpecifierSyntax.init(self)
-    }
-
-    /// Force-casts the current syntax node to `LifetimeTypeSpecifierSyntax`.
-    ///
-    /// - Returns: An instance of `LifetimeTypeSpecifierSyntax`.
-    /// - Warning: This function will crash if the cast is not possible. Use `as` to safely attempt a cast.
-    /// - Note: Requires experimental feature `nonescapableTypes`.
-    #if compiler(>=5.8)
-    @_spi(ExperimentalLanguageFeatures)
-    #endif
-    public func cast(_ syntaxType: LifetimeTypeSpecifierSyntax.Type) -> LifetimeTypeSpecifierSyntax {
-      return self.as(LifetimeTypeSpecifierSyntax.self)!
-    }
-  }
+  public typealias Element = SimpleTypeSpecifierSyntax
```

Comment:
This is an API-breaking change because clients might be switching over the element enum and using the `simpleTypeSpecifier` case to extract the value. It’s a little unfortunate that we did release swift-syntax 600 with `TypeSpecifierListSyntax.Element` as a `SyntaxChildChoices` and now need to undo it again.

@rintaro @hamishknight @bnbarham Do you think we should take the API breakage again or should we somehow keep `Element: SyntaxChildChoices` even if it only has a single case?

Reply:
@rintaro and I just discussed this and decided that what you did, ie. having a `SyntacChildChoices` with a single element, here is the best choice. It is one less API break and set us up in case we want to add type specifiers with detail in the future.

Reply:
Sorry, I mixed up what we wanted and what we have here. `TypeSpecifierListSyntax. Element` should continue being a `SyntaxChildChoices` enum, just with a single case and not change to `SimpleTypeSpecifierSyntax` in order to maintain API compatibility.